lib/combinePrimitives.js

```diff
@@ -0,0 +1,291 @@
+                currentGroup.push(nextPrimitive);
+                primitives.splice(i, 1);
+                i--;
+            }
+        }
+        primitiveGroups.push(currentGroup);
+    }
+
+    for (var j = 0; j < primitiveGroups.length; j++) {
+        combinedPrimitives.push(combinePrimitiveGroup(gltf, meshId, primitiveGroups[j]));
+    }
+
+    return combinedPrimitives;
+}
+
+
+//Checks if two primitives can be combined based on their attributes, materials, and modes.
+function readyToCombine(a, b) {
+    if (a.material !== b.material || a.mode !== b.mode || defined(a.indices) !== defined(b.indices)) {
+        return false;
+    }
+
+    var aKeys = Object.keys(a.attributes);
+    var bKeys = Object.keys(b.attributes);
+    if (aKeys.length !== bKeys.length) {
+        return false;
+    }
+
+    aKeys.sort();
+    bKeys.sort();
+    for (var i = 0; i < aKeys.length; i++) {
+        if (aKeys[i] !== bKeys[i]) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+//Creates a single primitive from a group of similar primitives.
+function combinePrimitiveGroup(gltf, meshId, primitiveGroup) {
+    var referencePrimitive = primitiveGroup[0];
+    if (primitiveGroup.length > 1) {
+        var newAttributes = {};
+        var newPrimitive = {
+            "material": referencePrimitive.material,
+            "mode": referencePrimitive.mode
+        };
+
+        //For each attribute, combine all referenced accessors into a single accessor.
+        if (defined(referencePrimitive.indices)) {
+            newPrimitive.indices = mergeAccessors(gltf, meshId, undefined, primitiveGroup, referencePrimitive);
+        }
+        var attributeKeys = Object.keys(referencePrimitive.attributes);
+        for (var i = 0; i < attributeKeys.length; i++) {
+            var attributeKey = attributeKeys[i];
+            newAttributes[attributeKey] = mergeAccessors(gltf, meshId, attributeKey, primitiveGroup, referencePrimitive);
+        }
+        newPrimitive.attributes = newAttributes;
+
+        return newPrimitive;
+    }
+
+    return referencePrimitive;
+}
+
+//Create a new accessor, bufferView, and buffer based on those referenced by the primitive group.
+//If the attributeType is undefined, we are merging an index accessor.
+function mergeAccessors(gltf, meshId, attributeType, primitiveGroup, referencePrimitive) {
+    var accessors = gltf.accessors;
+    var bufferViews = gltf.bufferViews;
+    var buffers = gltf.buffers;
+    var source = new Buffer(0);
+    var referenceAccessor = defined(attributeType) ? accessors[referencePrimitive.attributes[attributeType]] : accessors[referencePrimitive.indices];
+    var byteStride = defaultValue(referenceAccessor.byteStride, 0);
+    var componentType = referenceAccessor.componentType;
+    var type = referenceAccessor.type;
+    var count = 0;
+    var target, max, min;
+    var maxIndex = -1;
+
+    var attributeKey = defaultValue(attributeType, 'INDEX');
+    var newBuffer = {

```

Comment:
I believe we have another stage that combines buffers, and compacts buffers to remove their unused parts.  Remind me, where is that?  Perhaps add a comment here to indicate that stage should be ran later in the pipeline after this since looking at this in isolation doesn't make that clear.

Feel free to add this comment directly in master or another PR since I am going to merge this.